readme.md

```diff
@@ -400,10 +400,11 @@ String.method ('trim', function {
 //Makes a method available to all functions, ONLY when it definitely does not already exist
 
 Function.prototype.method (methodName, func) {
-	if (!this.prototype[methodName]){
-		this.prototype[methodName] = func;
-		return this;
+	if (this.prototype[methodName]) {
+    		throw new Error('Method with name ' + methodName + ' is already defined.');
```

Comment:
why would you `throw` an `error` if the method _is_ defined?
sure, we want to avoid overwriting the method, but we don't want to `throw` just for _testing_ if it exists.

Reply:
@nelsonic here's my way of thinking.

Someone (a Consumer) is calling this function to declare a new method, right? If this method is NOT set up successfully, some other code in Consumer _will break_ because it relies on this new method, otherwise why even bother setting up a method. This is a bad situation and we want to know about it as soon as it's detected so that the system does not break somewhere downstream.

```
// `target` has some default implementation defined somewhere else (say third-party library)
target.foo = function (a, b) { return a + b; };

// In my code I'm trying to define a method which happens to have the same name.
target.method('foo', function (a, b) { return a * b; });

target.foo(3, 5);   // Produces 8 instead of 15, because the original implementation was never overriden
```

On another hand, overriding the existing method is _often_ not good too. It is catastrophic if by overriding the existing method we break the original contract/semantics of the method. The only case I can think of where overriding is good, would be "monkey-patching", i.e. keep the interface, contract, and semantics the same, plus may be add some safe side effects.

Either the first or the second scenario should result in an error. I think, the `method` function should fail with an error when it was unable to fulfill its purpose which is _set up a method_. In original implementation it's situation where a functionName is already used in a prototype.

Reply:
I would